### PR TITLE
Switch all bignumber to decimal.js

### PR DESCRIPTION
**Pull request by tiagofilipenunes**
#533

### CODE REVIEW

**Comment by tiagofilipenunes** [comment]
Was careful with:

new BigNumber(undefined) and new BigNumber('') return 'NaN'
new Decimal(undefined) and new Decimal("") returns an error. new Decimal('NaN') returns 'NaN'

**Review by GrandSchtroumpf — APPROVED** [review]
Code looks good to me.

File: src/components/common/TokenInputField/Slippage.tsx [hunk 1]
```diff
@@ -16,14 +16,14 @@ export const Slippage: FC<{ slippage: BigNumber }> = ({ slippage }) => {
       <div className="flex-end flex gap-5">
         <div
           className={`ml-4 ${
-            slippage.gte(new BigNumber(-3)) && slippage.lte(new BigNumber(0))
+            slippage.gte(new Decimal(-3)) && slippage.lte(0)
```
Comment (pingustar) [inline src/components/common/TokenInputField/Slippage.tsx:19]: You can just pass `-3` here

File: src/components/strategies/edit/EditStrategyBudgetContent.tsx [hunk 2]
```diff
@@ -60,16 +60,16 @@ export const EditStrategyBudgetContent = ({
   } = useLocation();
 
   const calculatedOrder0Budget = !!order0.budget
-    ? new BigNumber(strategy.order0.balance)?.[
+    ? new Decimal(strategy.order0.balance || 0)?.[
```
Comment (pingustar) [inline src/components/strategies/edit/EditStrategyBudgetContent.tsx:63]: there seems to be something weird here.

We check for `!!order0.budget` but then reference `strategy.order0.balance` - the OR ` || 0` statement shouldnt be necessary.

**Review by pingustar — COMMENTED** [review]
Overall looks good. 

I am seeing more than I expected of the ` || 0` fallback to zero. Not sure if its a problem but seems odd.

**Review by tiagofilipenunes — COMMENTED** [review]
Merge main branch

**Review by zavelevsky — APPROVED** [review]
LGTM

**Pull request merged**